paddle/framework/tensor.cc

```diff
@@ -0,0 +1,32 @@
+#include "paddle/framework/tensor.h"
+
+namespace paddle {
+namespace framework {
+
+int Tensor::Rank() const { return arity(dims_); }
+
+int Tensor::Numel() const { return product(dims_); }
+
+void Tensor::Resize(const DDim& dims) {
+  dims_ = dims;
+  return;
+}
+
+void Tensor::Reshape(const DDim& dims) {
+  if (product(dims) != product(dims_)) {
+    // TODO: error: "Reshape() can not change tensor's numel".
+  }
+  dims_ = dims;
+  return;
+}
+
+const std::shared_ptr<Tensor::Placeholder>& Tensor::Holder() const {
```

Comment:
I think that users don't need to and shouldn't know the concept "holder", which is defined as a private type inside Tensor in my design.